### PR TITLE
Refinements in server logging

### LINKED ISSUE #1607: Refinements to server logging to consider
Here is what current master, at the time this was filed (see https://github.com/distributed-system-analysis/pbench/commit/72ca85a211ee245ad623e8dd5b51ea3f8a557582), generates for logs, once a minute, on a pbench server when it is idle (edited to group them together):

```
Apr 18 12:00:01 pbench.example.com CROND[17266]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-unpack-tarballs-small.lock /opt/pbench-server/bin/pbench-unpack-tarballs small)
Apr 18 12:00:02 pbench.example.com python3[17929]: 2020-04-18T12:00:02.647469 INFO 17929 140648039032640 pbench-unpack-tarballs-small.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17929]: 2020-04-18T12:00:02.650709 INFO 17929 140648039032640 pbench-unpack-tarballs-small.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17271]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-unpack-tarballs-medium.lock /opt/pbench-server/bin/pbench-unpack-tarballs medium)
Apr 18 12:00:02 pbench.example.com python3[17866]: 2020-04-18T12:00:02.533866 INFO 17866 139665604282176 pbench-unpack-tarballs-medium.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17866]: 2020-04-18T12:00:02.538225 INFO 17866 139665604282176 pbench-unpack-tarballs-medium.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17262]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-unpack-tarballs-large.lock /opt/pbench-server/bin/pbench-unpack-tarballs large)
Apr 18 12:00:02 pbench.example.com python3[17864]: 2020-04-18T12:00:02.527925 INFO 17864 140482201528128 pbench-unpack-tarballs-large.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17864]: 2020-04-18T12:00:02.532284 INFO 17864 140482201528128 pbench-unpack-tarballs-large.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17263]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-unpack-tarballs-huge.lock /opt/pbench-server/bin/pbench-unpack-tarballs huge)
Apr 18 12:00:02 pbench.example.com python3[17867]: 2020-04-18T12:00:02.534191 INFO 17867 139640844646208 pbench-unpack-tarballs-huge.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17867]: 2020-04-18T12:00:02.537896 INFO 17867 139640844646208 pbench-unpack-tarballs-huge.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17265]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-index-tool-data.lock /opt/pbench-server/bin/pbench-index --tool-data)
Apr 18 12:00:02 pbench.example.com python3[17379]: 2020-04-18T12:00:02.035826 INFO 17379 140181928503104 pbench-index-tool-data.pbench-index main 168 -- pbench-index-tool-data.run-2020-04-18T12:00:02-UTC: starting
Apr 18 12:00:02 pbench.example.com python3[17379]: 2020-04-18T12:00:02.039082 INFO 17379 140181928503104 pbench-index-tool-data.pbench-index main 212 -- No tar balls found that need processing


Apr 18 12:00:01 pbench.example.com CROND[17270]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-index.lock /opt/pbench-server/bin/pbench-index)
Apr 18 12:00:02 pbench.example.com python3[17375]: 2020-04-18T12:00:02.035743 INFO 17375 139875660703552 pbench-index.pbench-index main 168 -- pbench-index.run-2020-04-18T12:00:02-UTC: starting
Apr 18 12:00:02 pbench.example.com python3[17375]: 2020-04-18T12:00:02.038968 INFO 17375 139875660703552 pbench-index.pbench-index main 212 -- No tar balls found that need processing


Apr 18 12:00:01 pbench.example.com CROND[17267]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-sync-satellite-STAGING.lock /opt/pbench-server/bin/pbench-sync-satellite satellite-staging)
Apr 18 12:00:02 pbench.example.com python3[18014]: 2020-04-18T12:00:02.847105 INFO 18014 140559911851840 pbench-sync-satellite.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[18014]: 2020-04-18T12:00:02.850929 INFO 18014 140559911851840 pbench-sync-satellite.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17269]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-prep-002.lock /opt/pbench-server/bin/pbench-server-prep-shim-002)
Apr 18 12:00:02 pbench.example.com python3[17491]: 2020-04-18T12:00:02.317033 INFO 17491 139692599502656 pbench-server-prep-shim-002.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17491]: 2020-04-18T12:00:02.321849 INFO 17491 139692599502656 pbench-server-prep-shim-002.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.00s, successes: 1, duplicates: 0, failures: 0, retries: 0)


Apr 18 12:00:01 pbench.example.com CROND[17272]: (pbench) CMD (flock -n /opt/pbench-server/lib/locks/pbench-dispatch.lock /opt/pbench-server/bin/pbench-dispatch)
Apr 18 12:00:02 pbench.example.com python3[17799]: 2020-04-18T12:00:02.497498 INFO 17799 140581525210944 pbench-dispatch.indexer update_templates 520 -- done templates (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, retries: 0)
Apr 18 12:00:02 pbench.example.com python3[17799]: 2020-04-18T12:00:02.503438 INFO 17799 140581525210944 pbench-dispatch.report post_status 223 -- posted status (end ts: 2020-04-18T12:00:02-UTC, duration: 0.01s, successes: 1, duplicates: 0, failures: 0, retries: 0)
```

Here are a list of behavioral changes that we should consider making:

 - [x] Modify pbench-index to emit it "start" message under DEBUG logging level
 - [x] Modify pbench-index to emit start and end times (and calculated duration) in one INFO message with a full summary of actions taken
 - [x] Modify all successful "update templates" logs under DEBUG logging level
 - [x] Modify all post_status logs to not include a success count of 1 when nothing happened, other than the successful post of that status message, and include a start & end time, along with duration in a single log message

### PR DESCRIPTION
**Pull request by riya-17**
Fixes #1607 

This PR includes some behavioral changes to logging messages

### CODE REVIEW

**Comment by riya-17** [comment]
>     * [ ]  Modify all post_status logs to not include a success count of 1 when nothing happened, other than the successful post of that status message, and include a start & end time, along with duration in a single log message

@portante the value of success is changing in update_templates. Should I keep the status of update_templates and post_status same and then make required changes or we can combine them together as well.
Would like to get your views on this point.

File: pbench/indexer.py [hunk 1]
```diff
@@ -511,14 +511,26 @@ def update_templates(self, es, target_name=None):
...
                 retries += _retries
-        self.logger.info(
-            "done templates (end ts: {}, duration: {:.2f}s,"
-            " successes: {:d}, retries: {:d})",
-            pbench.tstos(end),
-            end - beg,
-            successes,
-            retries,
-        )
+        if retries > 0:
```
Comment (portante) [inline pbench/indexer.py:514]: You can make this a bit more compact via:

```
log_action = self.logger.warning if retries > 0 else self.logger.debug
log_action(
    "done templates (start ts: {}, end ts: {}, duration: {:.2f}s,"
    " successes: {:d}, retries: {:d})",
    pbench.tstos(beg),
    pbench.tstos(end),
    end - beg,
    successes,
    retries,
)
```

File: pbench/report.py [hunk 2]
```diff
@@ -210,18 +210,51 @@ def _es_payload_gen(_payload_gen):
...
-                    " successes: {:d}, duplicates: {:d}, failures: {:d},"
-                    " retries: {:d})",
-                    pbench.tstos(end),
-                    end - beg,
-                    successes,
-                    duplicates,
-                    failures,
-                    retries,
-                )
+                if failures > 0:
```
Comment (portante) [inline pbench/report.py:213]: A similar compaction can be made here as well:

```
if failures > 0:
    log_action = self.logger.error
elif duplicates > 0 or retries > 0:
    log_action = self.logger.warning
else:
    assert (...), "..."
    log_action = self.logger.debug
log_action( ... )
```

**Comment by portante** [comment]
> Is this supposed to be on top of br69 or of master?

Definitely `master`.

> Regardless, unit tests are failing, I presume because that are changes in the output but no changes in the gold files.

Yes, the unit test failures need to be addressed as well.

**Comment by riya-17** [comment]
@portante @ndokos

File: server/lib/pbench/report.py [hunk 3]
```diff
@@ -210,11 +210,23 @@ def _es_payload_gen(_payload_gen):
...
+                else:
+                    assert (
+                        successes >= 1
+                        and duplicates == 0
+                        and failures == 0
+                        and retries == 0
+                    ), "Logic Bomb!"
+                    log_action = self.logger.debug
+                log_action(
+                    " posted status (start ts: {}, end ts: {}, duration: {:.2f}s,"
```
[the remark this answers is not in the document]
Comment (riya-17) [reply server/lib/pbench/report.py]: ok sure

[the remark this answers is not in the document]
Comment (riya-17) [reply server/lib/pbench/report.py]: ok sure

**Pull request merged**